### PR TITLE
renamed BlockSoundGroup and Material related to Moss blocks

### PR DESCRIPTION
**Pull request by Redd56**
simple as that, have a great day with the new name if yall merge it 
refrence issue #2228

### CODE REVIEW

File: mappings/net/minecraft/block/Material.mapping [hunk 1]
```diff
@@ -11,7 +11,7 @@ CLASS net/minecraft/class_3614 net/minecraft/block/Material
 	FIELD field_15919 PORTAL Lnet/minecraft/class_3614;
 		COMMENT Material for the various portal blocks.
 	FIELD field_15920 WATER Lnet/minecraft/class_3614;
-	FIELD field_15921 UNUSED_PLANT Lnet/minecraft/class_3614;
+	FIELD field_15921 MOSS Lnet/minecraft/class_3614;
 		COMMENT Not in use, but has foliage color.
```
Comment (liach) [inline mappings/net/minecraft/block/Material.mapping]: If you don't mind, consider update this javadoc as well

Comment (Redd56) [reply mappings/net/minecraft/block/Material.mapping]: ah didnt notice that thanks for the bump

**Review by Redd56 — COMMENTED** [review]
yep looks good

**Comment by liach** [comment]
> 
> 
> yep looks good

If you changed locally, remember to push your changes. We cannot see your changes here yet.

**Comment by Redd56** [comment]
> > yep looks good
> 
> If you changed locally, remember to push your changes. We cannot see your changes here yet.

wait where is the javadoc. hol up let me find it, io thought you where refrencing it, give me a few to find it

**Comment by liach** [comment]
If you use enigma, you can navigate to that element, press control (command on mac) + d to open javadoc ui. or right click on the element and click edit javadocs

**Comment by Redd56** [comment]
> If you use enigma, you can navigate to that element, press control (command on mac) + d to open javadoc ui. or right click on the element and click edit javadocs

what is engima. i was just using idea

**Comment by Redd56** [comment]
also im in the discord if ya wanna go more in depth about what enigma is and how to use it

File: mappings/net/minecraft/block/Material.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/block/Material.mapping]:
```suggestion
		COMMENT Material for the moss block. Has foliage color.
```

**Comment by Redd56** [comment]
well i found enigma but like, its yelling and not launching, wonderful

**Comment by liach** [comment]
You can probably just take https://github.com/FabricMC/yarn/pull/2234#discussion_r605331464

**Comment by Redd56** [comment]
> You can probably just take [#2234 (comment)](https://github.com/FabricMC/yarn/pull/2234#discussion_r605331464)

SDLKFJSDLKFJSDKLFSHTHESLJ IT WASNT SHOWING THAT ANYWHERE WHY DID THAT ONLY JUST SHOW UP NOW GITHUB!

**Comment by Redd56** [comment]
ima kick githubs ass

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/block/Material.mapping [hunk 2]
```diff
@@ -11,8 +11,8 @@ CLASS net/minecraft/class_3614 net/minecraft/block/Material
 	FIELD field_15919 PORTAL Lnet/minecraft/class_3614;
 		COMMENT Material for the various portal blocks.
 	FIELD field_15920 WATER Lnet/minecraft/class_3614;
-	FIELD field_15921 UNUSED_PLANT Lnet/minecraft/class_3614;
-		COMMENT Not in use, but has foliage color.
+	FIELD field_15921 MOSS Lnet/minecraft/class_3614;
+		COMMENT Material for the moss block. Has foliage color.
```
Comment (dexman545) [inline mappings/net/minecraft/block/Material.mapping]: MapColor is per blockstate, please don't directly reference the default MapColor a material uses.

**Review by dexman545 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/block/Material.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/block/Material.mapping]: This is the old docs, so should be changed in another pr i think

Comment (Redd56) [reply mappings/net/minecraft/block/Material.mapping]: well this isnt MapColor. this is Material.  moss uses Green as the MapColor
unless im missunderstanding

Comment (dexman545) [reply mappings/net/minecraft/block/Material.mapping]: Yes, I'm asking for "Has foliage color" portion to be removed. It was originally added as at the time it was really the only information we had on it, but now we know what it is used for - a Material identical to GOURDS used only for moss.

Comment (dexman545) [reply mappings/net/minecraft/block/Material.mapping]: For reference, "foliage color" is what "dark green" MapColor was originally called. Apparently I forgot to update the JD when I renamed it.

Comment (liach) [reply mappings/net/minecraft/block/Material.mapping]:
```suggestion
		COMMENT Material for the moss block.
```
Guess we will do this

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/block/Material.mapping [hunk 3]
```diff
@@ -11,8 +11,8 @@ CLASS net/minecraft/class_3614 net/minecraft/block/Material
 	FIELD field_15919 PORTAL Lnet/minecraft/class_3614;
 		COMMENT Material for the various portal blocks.
 	FIELD field_15920 WATER Lnet/minecraft/class_3614;
-	FIELD field_15921 UNUSED_PLANT Lnet/minecraft/class_3614;
-		COMMENT Not in use, but has foliage color.
+	FIELD field_15921 MOSS_BLOCK Lnet/minecraft/class_3614;
+		COMMENT Material for the moss block.
```
Comment (dexman545) [inline mappings/net/minecraft/block/Material.mapping]:
```suggestion
		COMMENT Material for full sized moss blocks.
```
To match the pattern set by `SNOW_BLOCK`

**Review by dexman545 — CHANGES_REQUESTED** [review]
Good besides a minor nitpick now

File: mappings/net/minecraft/block/Material.mapping (hunk 3 above)
Comment (Redd56) [reply mappings/net/minecraft/block/Material.mapping]: ah true

**Review by dexman545 — APPROVED** [review]

**Comment by Redd56** [comment]
fully resolved to what i assume is now perfection, any more nitpicks cause i love to fix them. like no joke its nice to get things perfect to me. Gotta get everything perfect the first time, dont wanna recommit changes into the branch later.

**Comment by Redd56** [comment]
give a few minutes i might have stuff to add to it
sound group sillynes was found

**Comment by liach** [comment]
@Redd56 open a new pr for that, or this pr may be delayed

**Comment by liach** [comment]
Ah, this change is close enough. Sure, this can go in.

**Comment by Redd56** [comment]
nvm saving for new pr

**Comment by Redd56** [comment]
ill do that one once this one is merged
unless you want it in now

**Comment by liach** [comment]
You can have it here. I thought you were like going to another package for some new stuff. We've had prs initially going for mapping advancement positioner ending up mapping like iceberg world generation

**Comment by Redd56** [comment]
um. thats kinda cursed how far they deviated... jesus.... 
anyway readded it back since its just barely within scope

**Review by YanisBft — APPROVED** [review]

**Pull request merged**